### PR TITLE
get card endpoint

### PR DESCRIPTION
**Pull request by muhammadharoun**

### CODE REVIEW

**Review by rawand1998 — APPROVED** [review]

**Review by mohammedsalah7 — APPROVED** [review]

File: server/database/connection.js [hunk 1]
```diff
@@ -16,7 +16,8 @@ switch (NODE_ENV) {
     dburl = TEST_DB_URL;
     break;
   default:
-    throw new Error('DB Connection Failed');
+    // throw new Error('DB Connection Failed');
+    dburl = DB_URL;
```
Comment (Mu7ammadAbed) [inline server/database/connection.js:20]: why? you shouldn't do this

File: server/routes/getCard.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const { getCardQuery } = require('../database/queries');
+
+const getCardHandler = (req, res) => {
+  const { userId } = req.body;
+  getCardQuery(parseInt(userId, 10))
+    .then((result) => res.json((result.rows)))
```
Comment (Mu7ammadAbed) [inline server/routes/getCard.js:6]:
```suggestion
    .then(({ rows }) => res.json(({ data: rows })))
```

File: server/routes/getCard.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+    .catch((err) => res.json({ err }));
```
Comment (Mu7ammadAbed) [inline server/routes/getCard.js:7]:
```suggestion
    .catch((err) => next(err));
```

File: server/routes/addUser.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ const addUserHandler = (req, res) => {
     hashPassword(password)
       .then((hashpass) => {
         addUserQuery(name, email, hashpass, isAdmin, imgUrl)
-          .then(() => res.json({ status: 'scusses' }))
+          .then(() => res.status(201).json({ status: 'scusses' }))
```
Comment (Mu7ammadAbed) [inline server/routes/addUser.js:16]:
```suggestion
          .then(() => res.status(201).json({ status: 'User created successfully' }))
```

File: server/routes/clearCard.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+const { clearCardQuery } = require('../database/queries');
+
+const clearCardHandler = (req, res) => {
+  const { userId } = req.body;
```
Comment (Mu7ammadAbed) [inline server/routes/clearCard.js:4]: This shouldn't be sent with the request body

File: server/routes/deleteCard.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+const { deleteCardQuery } = require('../database/queries');
+
+const deleteCardHandler = (req, res) => {
+  const { userId, productId } = req.body;
```
Comment (Mu7ammadAbed) [inline server/routes/deleteCard.js:4]: The userId here shouldn't be sent from the request body

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

**Review by mohammedsalah7 — DISMISSED** [review]

File: server/routes/index.js [hunk 7]
```diff
@@ -5,12 +5,15 @@ const getCardHandler = require('./getCard');
...
 const router = express.Router();
 
 router.post('/singup', addUserHandler);
 router.post('/addcard', addCardHandler);
-router.post('/getcard', getCardHandler);
-router.post('/clearcard', getCardHandler);
-router.post('/deletecard', getCardHandler);
+router.get('/getcard/:userId', getCardHandler);
+router.delete('/clearcard/:userId', clearCardHandler);
+router.delete('/deletecard/:userId/:productId', deleteCardHandler);
```
Comment (Mu7ammadAbed) [inline server/routes/index.js:16]: You shouldn't send the userId for these routes by params, you should take it from the cookies after verifying, even if you still didn't handle this middleware properly you should add it and make your controller prepared for it
```suggestion
router.get('/getcard/:userId', getCardHandler);
router.delete('/clearcard', clearCardHandler);
router.delete('/deletecard/:productId', deleteCardHandler);
```

File: server/routes/addCard.js [hunk 8]
```diff
@@ -10,8 +10,8 @@ const addCardHandler = (req, res) => {
   });
   if (error === undefined) {
     addCardQuery(userId, productId, quantity)
```
Comment (Mu7ammadAbed) [inline server/routes/addCard.js:12]: You need to change your functions and queries name from `card` to `cart`
```suggestion
    addCartQuery(userId, productId, quantity)
```

File: server/routes/clearCard.js [hunk 9]
```diff
@@ -0,0 +1,9 @@
+const { clearCardQuery } = require('../database/queries');
+
+const clearCardHandler = (req, res) => {
```
Comment (Mu7ammadAbed) [inline server/routes/clearCard.js:3]:
```suggestion
const { clearCartQuery } = require('../database/queries');

const clearCartHandler = (req, res) => {
```

File: server/routes/clearCard.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
...
+  const { userId } = req.params;
+  clearCardQuery(parseInt(userId, 10))
```
Comment (Mu7ammadAbed) [inline server/routes/clearCard.js:5]:
```suggestion
  clearCartQuery(parseInt(userId, 10))
```

File: server/routes/clearCard.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+    .then(() => res.status(202).json({ status: 'cart clear successfully' }))
+    .catch((err) => res.json({ err }));
+};
+module.exports = clearCardHandler;
```
Comment (Mu7ammadAbed) [inline server/routes/clearCard.js:9]:
```suggestion
module.exports = clearCartHandler;
```

File: server/routes/deleteCard.js [hunk 12]
```diff
@@ -0,0 +1,9 @@
+const { deleteCardQuery } = require('../database/queries');
+
+const deleteCardHandler = (req, res) => {
+  const { userId, productId } = req.params;
+  deleteCardQuery(parseInt(userId, 10), parseInt(productId, 10))
+    .then(() => res.status(202).json({ status: 'product delete successfully' }))
+    .catch((err) => res.json({ err }));
+};
+module.exports = deleteCardHandler;
```
Comment (Mu7ammadAbed) [inline server/routes/deleteCard.js:9]:
```suggestion
const { deleteCartQuery } = require('../database/queries');

const deleteCartHandler = (req, res) => {
  const { userId, productId } = req.params;
  deleteCartQuery(parseInt(userId, 10), parseInt(productId, 10))
    .then(() => res.status(202).json({ status: 'product delete successfully' }))
    .catch((err) => res.json({ err }));
};
module.exports = deleteCartHandler;
```

File: server/routes/getCard.js [hunk 13]
```diff
@@ -0,0 +1,10 @@
+const { getCardQuery } = require('../database/queries');
+
+const getCardHandler = (req, res, next) => {
+  const { userId } = req.params;
+  getCardQuery(parseInt(userId, 10))
+    .then(({ rows }) => res.json(({ data: rows })))
+    .catch((err) => res.json({ err }))
+    .catch((err) => next(err));
+};
+module.exports = getCardHandler;
```
Comment (Mu7ammadAbed) [inline server/routes/getCard.js:10]:
```suggestion
const { getCartQuery } = require('../database/queries');

const getCartHandler = (req, res, next) => {
  const { userId } = req.params;
  getCartQuery(parseInt(userId, 10))
    .then(({ rows }) => res.json(({ data: rows })))
    .catch((err) => res.json({ err }))
    .catch((err) => next(err));
};
module.exports = getCartHandler;
```

File: server/routes/index.js [hunk 14]
```diff
@@ -1,10 +1,19 @@
 const express = require('express');
 const addUserHandler = require('./addUser');
 const addCardHandler = require('./addCard');
-const {getProduct} = require('../controllers/index')
+const getCardHandler = require('./getCard');
+const deleteCardHandler = require('./deleteCard');
+const clearCardHandler = require('./clearCard');
```
Comment (Mu7ammadAbed) [inline server/routes/index.js:6]:
```suggestion
const getCartHandler = require('./getCart');
const deleteCartHandler = require('./deleteCart');
const clearCartHandler = require('./clearCart');
```

**Review by rawand1998 — APPROVED** [review]

File: server/controllers/loginController.js [hunk 15]
```diff
@@ -19,6 +19,7 @@ exports.loginPost = async (req, res) => {
 
   // create token and store it in cookies
   const { id, name } = user.rows[0];
+  console.log(id, name);
```
Comment (Mu7ammadAbed) [inline server/controllers/loginController.js:22]:
```suggestion
```

File: server/routes/clearCart.js [hunk 16]
```diff
@@ -0,0 +1,9 @@
+const { clearCartQuery } = require('../database/queries');
+
+const clearCartHandler = (req, res) => {
+  const userId = req.user.id;
```
Comment (Mu7ammadAbed) [inline server/routes/clearCart.js:4]:
```suggestion
  const { id: userId } = req.user;
```

File: server/routes/deleteCart.js [hunk 17]
```diff
@@ -0,0 +1,12 @@
+const { deleteCartQuery } = require('../database/queries');
+
+const deleteCartHandler = (req, res) => {
+  const userId = req.user.id;
```
Comment (Mu7ammadAbed) [inline server/routes/deleteCart.js:4]:
```suggestion
  const { id: userId } = req.user;
```

File: server/routes/getCart.js [hunk 18]
```diff
@@ -0,0 +1,10 @@
+const { getCartQuery } = require('../database/queries');
+
+const getCartHandler = (req, res, next) => {
+  const userId = req.user.id;
```
Comment (Mu7ammadAbed) [inline server/routes/getCart.js:4]:
```suggestion
  const { id: userId } = req.user;
```

File: server/database/queries/getCartQuery.js [hunk 19]
```diff
@@ -1,5 +1,11 @@
 const connection = require('../connection');
 
-const getCartQuery = (userId) => connection.query(`select * FROM cart where user_id = ${userId};`);
+const getCartQuery = (userId) => {
+  const sql = {
+    text: 'select * FROM cart where user_id = $1;',
```
Comment (Mu7ammadAbed) [inline server/database/queries/getCartQuery.js:5]:
```suggestion
    text: 'SELECT * FROM cart WHERE user_id = $1;',
```

File: server/database/queries/deleteCartQuery.js [hunk 20]
```diff
@@ -1,5 +1,11 @@
 const connection = require('../connection');
 
-const deleteCartQuery = (userId, productId) => connection.query(`delete from cart where user_id =${userId} AND product_id = ${productId};`);
+const deleteCartQuery = (userId, productId) => {
+  const sql = {
+    text: 'delete from cart where user_id =$1 AND product_id = $2;',
```
Comment (Mu7ammadAbed) [inline server/database/queries/deleteCartQuery.js:5]:
```suggestion
    text: 'DELETE FROM cart WHERE user_id =$1 AND product_id = $2;',
```

File: server/database/queries/clearCartQuery.js [hunk 21]
```diff
@@ -1,5 +1,11 @@
 const connection = require('../connection');
 
-const clearCartQuery = (userId) => connection.query(`delete from cart where user_id = ${userId};`);
+const clearCartQuery = (userId) => {
+  const sql = {
+    text: 'delete from cart where user_id = $1;',
```
Comment (Mu7ammadAbed) [inline server/database/queries/clearCartQuery.js:5]:
```suggestion
    text: 'DELETE FROM cart WHERE user_id = $1;',
```

**Review by rawand1998 — DISMISSED** [review]

**Review by NourAdawi — DISMISSED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Pull request merged**